crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs

```diff
@@ -484,13 +475,56 @@ impl Analyzer<'_, '_> {
                     let mut analyzer = analyzer.with_ctx(ctx);
 
                     let result = match type_ann {
-                        Some(ty) if will_skip_this => {
+                        Some(ty) if rhs_is_arrow || !right_function_declared_this => {
+                            let mut ty = ty.clone();
+                            if lhs_declared_this {
+                                if let Type::Function(stc_ts_types::Function { params, .. }) = ty.normalize_mut() {
```

Comment:
Please use `if let Some(stc_ts_types::Function { params, .. }) = ty.as_fn_type_mut()`.

Reply:
This un-freeze `ty` only if `ty` is `Type::Function`

Reply:
i fixed it